integrations/event-handler/state.go

```diff
@@ -238,7 +245,91 @@ func (s *State) SetSessionIndex(id string, index int64) error {
 	return s.dv.Write(sessionPrefix+id, b)
 }
 
-// RemoveSession removes session from the state
+// SetMissingRecording writes the session with missing recording into state.
+func (s *State) SetMissingRecording(sess session, attempt int) error {
+	b, err := json.Marshal(missingRecording{
+		Index:     sess.Index,
+		Attempt:   attempt,
+		Timestamp: sess.UploadTime,
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if err := s.dv.Write(missingRecordingPrefix+sess.ID, b); err != nil {
```

Comment:
Should we use `WriteStream` instead and set the sync to true?
I will probably take longer but it brings the durability guarantee.

Reply:
Not really. Maybe we should replace everything with `WriteStream` in a follow up PR and keep this one as is for the sake of consistency.